frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FC, MouseEvent, useState } from 'react';
+import { withStyles, WithStyles } from '@material-ui/core/styles';
+import { ListItem, ListItemText, Collapse } from '@material-ui/core';
+import { ExpandLess, ExpandMore } from '@material-ui/icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMinusSquare } from '@fortawesome/free-solid-svg-icons';
+import { faSquare } from '@fortawesome/free-regular-svg-icons';
+
+import { tagListItemStyles } from './styles/TagListStyles';
+import TagViewFeatureList from './TagViewFeatureList';
+import Tag from 'models/Tag';
+import Status from 'models/Status';
+import TagAvatar from './TagAvatar';
+
+interface Props extends WithStyles {
+  loggedInUserName: string;
+  isEditMode: boolean;
+  isAssignedTagsView: boolean;
+  tag: Tag;
+  restId: string;
+  selectedFeatureId: string;
+  selectedStatus: Status;
+  handleFeatureSelected(): void;
+  handleTagAssigned(restId: string, name: string, loggedInUserName: string, userName?: string): void;
+  handleWarningShow(restId: string, name: string, loggedInUserName: string, userName?: string): void;
+  handleTagIgnore(productId: string, name: string): void;
+}
+
+const TagListItem: FC<Props> = ({
+  loggedInUserName,
+  isEditMode,
+  isAssignedTagsView,
+  tag,
+  selectedFeatureId,
+  restId,
+  selectedStatus,
+  handleFeatureSelected,
+  handleTagAssigned,
+  handleWarningShow,
+  handleTagIgnore,
+  classes,
+}) => {
+  const [expanded, setExpanded] = useState(false);
+  const featureList = tag.features.filter(feature => selectedStatus[feature.calculatedStatus]);
+  const { userName, name, isIgnored } = tag;
+
+  const onAvatarClick = (event: MouseEvent): void => {
+    event.stopPropagation();
+    if (userName && userName !== loggedInUserName) {
+      handleWarningShow(restId, name, loggedInUserName, userName);
+    } else {
+      handleTagAssigned(restId, name, loggedInUserName, userName);
```

Comment:
handleTagAssigned has the same condition in it as here. Would it be clearer to put all the logic in one place here so the handlers don't have to make any decisions? eg something like:
if (!username) assignToMe
else if (userName === me) unassign
else showWarning.

Reply:
Yeah definitely, this was copy paste so I can clean it up further